components/team_controller/team_controller.tsx

```diff
@@ -0,0 +1,258 @@
+import {makeAsyncComponent} from 'components/async_load';
+import ChannelController from 'components/channel_layout/channel_controller';
+
+import LocalStorageStore from 'stores/local_storage_store';
+
+import type {OwnProps, PropsFromRedux} from './index';
+
+const BackstageController = makeAsyncComponent('BackstageController', lazy(() => import('components/backstage')));
+const Pluggable = makeAsyncComponent('Pluggable', lazy(() => import('plugins/pluggable')));
+
+const WAKEUP_CHECK_INTERVAL = 30000; // 30 seconds
+const WAKEUP_THRESHOLD = 60000; // 60 seconds
+const UNREAD_CHECK_TIME_MILLISECONDS = 10000;
+
+declare global {
+    interface Window {
+        isActive: boolean;
+    }
+}
+
+type Props = PropsFromRedux & OwnProps;
+
+function TeamController(props: Props) {
+    const history = useHistory();
+    const {team: teamNameParam} = useParams<Props['match']['params']>();
+
+    const [team, setTeam] = useState<Team | null>(getTeamFromTeamList(props.teamsList, teamNameParam));
+
+    const blurTime = useRef(new Date().getTime());
+    const lastTime = useRef(Date.now());
+
+    function handleFocus() {
+        if (props.selectedThreadId) {
+            window.isActive = true;
+        }
+        if (props.currentChannelId) {
+            window.isActive = true;
+            props.markChannelAsReadOnFocus(props.currentChannelId);
+        }
+
+        const currentTime = (new Date()).getTime();
+        if ((currentTime - blurTime.current) > UNREAD_CHECK_TIME_MILLISECONDS && props.currentTeamId) {
+            if (props.graphQLEnabled) {
+                props.fetchChannelsAndMembers(props.currentTeamId);
+            } else {
+                props.fetchMyChannelsAndMembersREST(props.currentTeamId);
+            }
+        }
+    }
+
+    function handleBlur() {
+        window.isActive = false;
+        blurTime.current = new Date().getTime();
+
+        if (props.currentUser) {
+            props.viewChannel('');
+        }
+    }
+
+    function handleKeydown(e: KeyboardEvent) {
+        if (e.shiftKey && cmdOrCtrlPressed(e) && isKeyPressed(e, Constants.KeyCodes.L)) {
+            const sidebar = document.getElementById('sidebar-right');
+            if (sidebar) {
+                if (sidebar.className.match('sidebar--right sidebar--right--expanded move--left')) {
+                    const replyTextbox = document.getElementById('reply_textbox');
+                    if (replyTextbox) {
+                        replyTextbox.focus();
+                    }
+                } else {
+                    const postTextbox = document.getElementById('post_textbox');
+                    if (postTextbox) {
+                        postTextbox.focus();
+                    }
+                }
+            }
+        }
+    }
+
+    useEffect(() => {
+        if (props.graphQLEnabled) {
+            props.fetchChannelsAndMembers();
+        } else {
+            props.fetchAllMyTeamsChannelsAndChannelMembersREST();
+        }
+    }, []);
```

Comment:
Just sanity check: We expect all the props used here to never change during the lifetime of the component?
(`graphQLEnabled`, `fetchChannelsAndMembers` and `fetchAllMyTeamsChannelsAndChannelMembersREST`)

Reply:
🆗 passed the `props.graphQLEnabled` in dependency